parcels/grid.py

```diff
@@ -199,17 +201,24 @@ def computeTimeChunk(self, f, time, signdt):
                 self.time = self.time_full
                 self.ti, _ = f.time_index(time)
                 periods = self.periods.value if isinstance(self.periods, c_int) else self.periods
-                if signdt == -1 and self.ti == 0 and (time - periods*(self.time_full[-1]-self.time_full[0])) == self.time[0]:
+                if signdt == -1 and self.ti == 0 and (time - periods*(self.time_full[-1]-self.time_full[0])) == self.time[0] and f.time_periodic:
                     self.ti = len(self.time)-2
                     periods -= 1
-
-                if self.ti > 0 and signdt == -1:
+                if signdt == -1 and self.ti > 0:
                     self.ti -= 1
                 if self.ti >= len(self.time_full) - 2:
                     self.ti = len(self.time_full) - 3
+
                 self.time = self.time_full[self.ti:self.ti+3]
                 self.tdim = 3
-                self.update_status = 'first_updated'
+                if prev_time_indices is None or len(prev_time_indices) != 3 or len(prev_time_indices) != len(self.time):
+                    self.update_status = 'first_updated'
+                elif functools.reduce(lambda i, j: i and j, map(lambda m, k: m == k, self.time, prev_time_indices), True) and len(prev_time_indices) == len(self.time):
```

Comment:
`lambda` functions are not very easy to read (we only have one other in Parcels); I thought that in python3 they were slowly being phased out and replaced by e.g. lost comprehension?

Reply:
this lambda function was, after google browse, the best and quicked option to check if the indices in both python arrays (prev_time_indices, self.time) are the same at the same position - also the quickest, cause all other versions depend on a `for i in range(len(self.time))` loop, which should never be done in performance-critical applications. I could rewrite all those simple loops with `itertools.islice()` to increase performance, but at this point it seemed the simplest and quickest option.

Generally, I do not like lambda expressions either cause they sometimes hide function calls as being variables. It's just that in this case, it makes sense due to performance.

Reply:
OK, is fine then